### ISSUE TITLE
Register new works with handle provider

### ISSUE THREAD

**Issue by mark-dce** [curationexperts/epigaea#23]
**ACCEPTANCE CRITERIA**

- [x] When the work is published, 
  - [x] if the work does not already have a handle, a new handle should be registered for the work
  - [x] if the work already has a handle stored in it's metadata, the handle should be verified and updated if it does not point to the correct URI for the work 
- [x] Handle registrations and updates should be performed as background jobs
- [x] Handle registrations and update errors should be logged
- [x] Any errors encountered registering or updating handles should be logged - see #24 for additional detail about how logs are accessed
- [ ] Handles should not be created until a work is published
- [x] There is a command line rake task to register handles for a work

**Comment by no-reply** [comment]
@mkorcy says Tufts has a demo handle prefix that we can use in production for the demos. My plan is to stub the handle service in the test suite, but it will be nice to have a real handle server in our local dev and prod enviornments.

**Comment by no-reply** [comment]
Can we clarify in what cases we want to register a handle?
  - For new drafted works?
  - For saved, unpublished works? or,
  - For published works only?

**Comment by mark-dce** [comment]
General case: for published works only. - I've updated the main acceptance criteria above accordingly

Special case: there should be a link on the toolbox - if I click "Update Handle" in the toolbox on a work that hasn't been published, it should register a new handle - I've updated the acceptance criteria in #95 accordingly.

**Comment by mark-dce** [comment]
EXISTING CODE  
https://github.com/TuftsUniversity/MIRA/blob/master/app/services/register_handle_service.rb   
https://github.com/TuftsUniversity/MIRA/blob/master/lib/job/register_handle.rb
https://github.com/TuftsUniversity/MIRA/blob/master/app/services/record_handle_service.rb
https://github.com/TuftsUniversity/MIRA/blob/master/app/services/handle_log_service.rb
https://github.com/TuftsUniversity/MIRA/blob/master/lib/tasks/handle.rake
https://github.com/TuftsUniversity/MIRA/blob/master/app/services/publish_service.rb#L14-L35


TESTS  
https://github.com/TuftsUniversity/MIRA/blob/master/spec/services/register_handle_service_spec.rb
https://github.com/TuftsUniversity/MIRA/blob/master/spec/job/register_handle_spec.rb
https://github.com/TuftsUniversity/MIRA/blob/master/spec/services/publish_service_spec.rb#L7-L47


ADDITIONAL CONTEXT  
https://github.com/TuftsUniversity/MIRA#handlenet

**Comment by no-reply** [comment]
> Handle registrations and updates should be logged
The existing [handle service](https://github.com/TuftsUniversity/MIRA/blob/master/app/services/register_handle_service.rb#L16-L20) only seems to log errors. My implementation in https://github.com/curationexperts/epigaea/compare/handle logs the errors, but not the registrations. 

@mark-dce: Can we confirm this requirement?

**Comment by no-reply** [comment]
> There is a command line rake task to register handles for a work

Does this task strictly register a Handle for an object? or:

  - does it register a handle contingent on one not already existing?
  - does it register a handle if one does not exist, and verify/update existing handles?

My sense is that the this task does the first bullet point above, but I'd like to confirm before implementing it. The backend pieces are in place for all of the above.

**Comment by no-reply** [comment]
Thanks all for answers during stand-up this morning. To document:
  - We removed the registration/update logging requirements.
  - We agreed that the handle registration task should be a simple register-if-not-present task, and that it should report existing handles.
  - We decided to move forward with Arks as handle extensions for now.

**Comment by no-reply** [comment]
@mkorcy I wonder if you can help me fill in the details for the development handle configuration at https://github.com/curationexperts/epigaea/compare/handle#diff-9a6e359b8f54ada5f2f923d04265c71c?

**Comment by mkorcy** [comment]
@no-reply i emailed @mark-dce about this I'll fwd it to you now.

**Comment by no-reply** [comment]
thanks!
